logs_analyzer/app/java8/src/main/java/com/databricks/apps/logs/ApacheAccessLog.java

```diff
@@ -121,11 +123,11 @@ public void setContentSize(long contentSize) {
       "^(\\S+) (\\S+) (\\S+) \\[([\\w:/]+\\s[+\\-]\\d{4})\\] \"(\\S+) (\\S+) (\\S+)\" (\\d{3}) (\\S+)";
   private static final Pattern PATTERN = Pattern.compile(LOG_ENTRY_PATTERN);
 
-  public static ApacheAccessLog parseFromLogLine(String logline) {
+  public static ApacheAccessLog parseFromLogLine(String logline) throws IOException {
```

Comment:
The API expects user to handle this exception, so it must be _Exception_ rather than RuntimeException. See the usage in LogAnalyzerAppMain:

            try {
                return Iterators.singletonIterator(ApacheAccessLog.parseFromLogLine(line));
            } catch (IOException e) {
                return Iterators.emptyIterator();
            }

With RuntimeException, the API becomes unclear for the user. So, I have chosen IOException as an Exception closely matching to the context: a handled exception when reading data.